debug/physmem.py

```diff
@@ -67,18 +67,21 @@ def get_all_segments(self):
 
     def dump_segment_freeq(self, idx, freeq, size):
         pages = tailq.collect_values(freeq, 'freeq')
-        return [[idx, size, as_hex(page['paddr']), as_hex(page['vaddr'])]
-                for page in pages]
+        return [[str(idx), str(size), as_hex(page['paddr']),
+                 as_hex(page['vaddr'])] for page in pages]
 
     def dump_segment_free_pages(self, idx, segment):
-        return [self.dump_segment_freeq(idx, segment['freeq'][q], 4 << q)
-                for q in range(16)]
+        helper = []
```

Comment:
I'm not sure how to incorporate chain() to make it work with ptable. If you think it can be done feel free to change it.